service/vehicle.go

```diff
@@ -0,0 +1,84 @@
+package service
+
+import (
+	"encoding/csv"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/edgarduran/Go-Dispatch-Bootcamp/models"
+)
+
+func GetAllVehicles() ([]models.Vehicle, error) {
+	f, err := os.Open("vehicles.csv")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer f.Close()
+
+	csvReader := csv.NewReader(f)
+	data, err := csvReader.ReadAll()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("var1 = %T\n", data)
+	var vehicleList []models.Vehicle
+	for _, vehicle := range data {
+		var rec models.Vehicle
+		for j, field := range vehicle {
+			if j == 0 {
+				id, _ := strconv.ParseInt(field, 10, 64)
+				rec.Id = id
+			} else if j == 1 {
+				rec.Make = field
+			} else if j == 2 {
+				rec.Model = field
+			}
+		}
+
+		vehicleList = append(vehicleList, rec)
+	}
+
+	return vehicleList, nil
+}
+
+func GetVehicleById(id int) ([]models.Vehicle, error) {
+	f, err := os.Open("vehicles.csv")
```

Comment:
Additional to what I wrote in the previous function, you should not be reading your file in every single method.

This file should be a dependency of a service struct, which will be used in a method, instead of a function, so you don't need to open your file in every single function you need it.

Receive in the constructor the name of the file, open it, convert it to an array of vehicles, and use that array for all your operations.